### PR TITLE
[SDK-2052] Refactor getTokenWithPopup tests

### PR DESCRIPTION
**Pull request by stevehobbsdev**
This cleans up the `getTokenWithPopup` tests to remove mocking at the highest level, and removes some of the old tests that test cache calls and whatnot, as they're no longer needed/relevant.

### CODE REVIEW

File: __tests__/Auth0Client.test.ts [hunk 1]
```diff
@@ -2055,6 +2057,98 @@ describe('Auth0Client', () => {
...
+        }
+      };
+
+      expect(await auth0.getTokenWithPopup({}, config)).toEqual(
+        TEST_ACCESS_TOKEN
+      );
+
+      expect(config.popup.location.href).toMatch(
+        /openid%20email%20read%3Aemail/
+      );
```
Comment (stevehobbsdev) [inline __tests__/Auth0Client.test.ts:2106]: `getTokenWithPopup` still does some of its own logic to pass in the right scopes, so I think this test is valuable.

File: __tests__/Auth0Client.test.ts [hunk 2]
```diff
@@ -2055,6 +2057,98 @@ describe('Auth0Client', () => {
...
+            href: ''
+          },
+          close: jest.fn()
+        }
+      };
+
+      await auth0.getTokenWithPopup(loginOptions, config);
+
+      expect(config.popup.location.href).toMatch(/other-audience/);
+      expect(config.popup.location.href).toMatch(/screen_hint/);
```
Comment (stevehobbsdev) [inline __tests__/Auth0Client.test.ts:2129]: `getTokenWithPopup` still does some of its own logic to pass in the right login options, so I think this test is valuable.

File: __tests__/Auth0Client.test.ts [hunk 3]
```diff
@@ -2055,6 +2057,98 @@ describe('Auth0Client', () => {
...
+          location: {
+            href: ''
+          },
+          close: jest.fn()
+        }
+      };
+
+      await auth0.getTokenWithPopup({}, config);
+
+      expect(config.popup.location.href).toMatch(/global-audience/);
```
Comment (stevehobbsdev) [inline __tests__/Auth0Client.test.ts:2148]: `getTokenWithPopup` still does some of its own logic to pass in the audience, so I think this test is valuable.

**Review by frederikprijck — APPROVED** [review]

**Comment by frederikprijck** [comment]
Looks good but it looks like Test Coverage dropped.

**Pull request merged**